codex/utils.nim

```diff
@@ -35,6 +36,24 @@ proc orElse*[A](a, b: Option[A]): Option[A] =
   else:
     b
 
+template findIt*(s, pred: untyped): untyped =
+  ## Returns the index of the first object matching a predicate, or -1 if no
+  ## object matches it.
+  runnableExamples:
+    type MyType = object
+      att: int
+
+    var s = @[MyType(att: 1), MyType(att: 2), MyType(att: 3)]
+    doAssert s.findIt(it.att == 2) == 1
+    doAssert s.findIt(it.att == 4) == -1
+
+  var index = -1
+  for i, it {.inject.} in enumerate(items(s)):
```

Comment:
meh, so lame that `enumerate`  only works on for loops, otherwise it could have been just chained in an expression: 

```nim
  arr.enumerate.map((i, e) => if e == pred: e else: -1).filterIt( it > -1 )
```

But this is then, the next best thing :)

slightly off-topic:

A consistent iterators/streams/collections with proper backpressure support, is something that I'd love to have in nim. I think @tbekas started something in `utils/asynciter.nim`, but still a long way to go.